### DISCUSSION TITLE
ucp driver not loading

### DISCUSSION THREAD

**Discussion by bellaabdelouahab** [bellaabdelouahab/Hotel-Management-System#16] [Q&A] [answered]
`java.lang.ClassCastException: oracle.ucp.jdbc.proxy.oracle$1ucp$1jdbc$1proxy$1oracle$1ConnectionProxy$2oracle$1jdbc$1internal$1OracleConnection$$$Proxy cannot be cast to oracle.jdbc.driver.OracleConnection`

**Comment by Khalid1G** [comment, marked as answer]
You can try type cast the UCP connection to oracle.jdbc.OracleConnection and then call below API to enqueue the message.
`public void enqueue(String queueName, oracle.jdbc.aq.AQEnqueueOptions opt, oracle.jdbc.aq.AQMessage mesg) throws SQLException;`
